README.md

```diff
@@ -14,25 +14,78 @@ It is work in progress, but functions well.
 | ![](before.png)                                |                                 ![](after.png) |
 
 # Supported Languages
-- [x] [C][tree-sitter-c]
-- [x] [C++][tree-sitter-cpp]
-- [ ] CSS
-- [x] [D][tree-sitter-d]
-- [x] [Diff][tree-sitter-diff]
-- [x] [Go][tree-sitter-go]
-- [x] [go.mod][tree-sitter-go-mod]
-- [ ] HTML
-- [x] [Javascript/JSX][tree-sitter-javascript]
-- [x] [Julia][tree-sitter-julia]
-- [x] [Lua][tree-sitter-lua]
-- [x] [Rust][tree-sitter-rust]
-- [x] [Zig][tree-sitter-zig]
-
-If you want more languages supported, open an issue.
+
+Any language supported by `tree-sitter` should (in theory) be supported as well.
+
+It is possible to use grammar installed locally or directly from their git repository. Additionally 
+it is possible to specify where to find the queries of the language as well as use sub path for projects
+containing more then one language.
+## Default grammars
+
+A few grammars have default configurations, pre-compiled parsers and custom queries.
+The supported languages are:  
+- [C][tree-sitter-c]
+- [C++][tree-sitter-cpp]
+- [Diff][tree-sitter-diff]
+- [Go][tree-sitter-go]
+- [go.mod][tree-sitter-go-mod]
+- [Javascript/JSX][tree-sitter-javascript]
+- [Julia][tree-sitter-julia]
+- [Lua][tree-sitter-lua]
+- [Rust][tree-sitter-rust]
+- [Zig][tree-sitter-zig]
+
+## Setting up a grammar
+
+It is possible to use a grammar for a specific language from the default
+configurations, from a git repository or from a local installation. To enable
+a grammar you should add a few configuration line to your lite-xl **init.lua**.
+
+### Examples
+
+```lua
+local egconf = require "plugins.evergreen.config"
+
+-- Activate a default grammar (see list above)
+-- it will automatically download and install the
+-- precompiled grammar and prefered queries.
+egconf.addGrammar {
+	lang = "c", -- name of the language
+	precompiled = true, -- use the precompiled version
+}
+
+-- Install a grammar from a git repository.
+-- It will automatically clone the repo, build the grammar
+-- and install it with the default queries.
+-- It requires **git** and **gcc**.
+egconf.addGrammar {
+	lang = "cue",
+	git = "https://github.com/eonpatapon/tree-sitter-cue.git" -- git repo
+  -- revision = "...", -- optional specific revision to use
+	-- supath = "...", -- sub path if the grammar is a part of a bigger repository (default ".")
+  -- queries = "...", -- specific folder where queries are found (default "queries/")
+  extensions = { "cue" }, -- list of extensions to match.
+                          -- it can be omitted if the language has the same name of the
+                          -- extension (like in this case)
+  -- filenames = {...}, -- list of filenames to match
```

Comment:
better explanation? what are the filenames matching for..

Reply:
The file names are  coming from the current implementation of evergreen and in particular at `languages.lua:30`.
But basically is for grammars that matches complete filenames and not only extensions, but I believe it would be better to add regular expression (such as `%.cue`) in order to first be more flexible (for example I have makesfile that could be matched  by something like this `Makefile.%`) and avoid the need to have the two separate fields. Moreover I think `lite-xl-lsp` uses something like that.

What do you think about it?

Reply:
Hi, I updated it with a single `filePatterns` entry where it is possible to specify any regex (as in `lite-xl-lsp:file_patterns`  (is really unfortunate to use different naming patterns...)). 
Please let me know what do you think about it.

Reply:
assuming this is the same as ignore_patters in lite xl, it's a good change.